internal/repos/syncer.go

```diff
@@ -491,6 +516,22 @@ func (s *Syncer) sync(ctx context.Context, svc *types.ExternalService, sourced *
 		panic("unreachable")
 	}
 
+	// PermsSyncer is available in enterprise mode only.
+	// Even in enterprise mode, only trigger repo permissions sync if we need to.
+	if s.PermsSyncer == nil || !scheduleRepoPermsSync {
+		return d, nil
+	}
+
+	// We expect the length of d.Added to be either 0 or 1. Ranging over it is simpler.
+	for _, r := range d.Added {
+		s.PermsSyncer.ScheduleRepos(ctx, r.ID)
+	}
+
+	// We expect the length of d.Modified to be either 0 or 1. Ranging over it is simpler.
+	for _, r := range d.Modified {
+		s.PermsSyncer.ScheduleRepos(ctx, r.ID)
+	}
```

Comment:
If we go to the extreme :trollface: 
```suggestion
	// We expect the length of d.Added and d.Modified to be either 0 or 1.
	// Ranging over it is simpler.
	for _, r := range append(d.Added, d.Modified...) {
		s.PermsSyncer.ScheduleRepos(ctx, r.ID)
	}
```

Reply:
I like this!